<launch>
  <arg name="initial_velocity_kmh" default="2.1" />
  <arg name="velcity_kmh_before_obstalcle" default="1.0" />
  <arg name="distance_before_obstacle" default="7.0" />
  <arg name="obstalce_radius_from_center_point" default="2.0" />
  <arg name="min_lateral_referencing_offset_for_avoidance" default="5.0" />
  <arg name="max_lateral_referencing_offset_for_avoidance" default="8.0"/>
  <arg name="diff_waypoints_cost_coef" default="1.0"/>
  <arg name="diff_last_waypoint_cost_coef" default="1.0"/>
  <arg name="jerk_cost_coef" default="0.0"/>
  <arg name="required_time_cost_coef" default="0.0"/>
  <arg name="comfort_acceleration_cost_coef" default="0.0"/>
  <arg name="lookahead_distance_per_kmh_for_reference_point" default="2.0"/>
  <arg name="converge_distance_per_kmh_for_stop" default="2.36"/>
  <arg name="linear_velocity_kmh" default="5.0"/>
  <arg name="only_testing_modified_global_path" default="false"/>
  <arg name="min_radius" default="1.6"/>
  <node pkg="frenet_planner" type="frenet_planner" name="frenet_planner" output="screen">
    <param name="initial_velocity_kmh"  value="$(arg initial_velocity_kmh)" />
    <param name="velcity_kmh_before_obstalcle"        value="$(arg velcity_kmh_before_obstalcle)" />
    <param name="distance_before_obstacle"   value="$(arg distance_before_obstacle)" />
    <param name="obstalce_radius_from_center_point"         value="$(arg obstalce_radius_from_center_point)" />
    <param name="min_lateral_referencing_offset_for_avoidance"   value="$(arg min_lateral_referencing_offset_for_avoidance)" />
    <param name="max_lateral_referencing_offset_for_avoidance"   value="$(arg max_lateral_referencing_offset_for_avoidance)" />
    <param name="diff_waypoints_cost_coef"   value="$(arg diff_waypoints_cost_coef)" />
    <param name="diff_last_waypoint_cost_coef"   value="$(arg diff_last_waypoint_cost_coef)" />
    <param name="jerk_cost_coef"   value="$(arg jerk_cost_coef)" />
    <param name="required_time_cost_coef"   value="$(arg required_time_cost_coef)" />
    <param name="comfort_acceleration_cost_coef"   value="$(arg comfort_acceleration_cost_coef)" />
    <param name="lookahead_distance_per_kmh_for_reference_point"   value="$(arg lookahead_distance_per_kmh_for_reference_point)" />
    <param name="converge_distance_per_kmh_for_stop"   value="$(arg converge_distance_per_kmh_for_stop)" />
    <param name="linear_velocity_kmh"   value="$(arg linear_velocity_kmh)" />
    <param name="only_testing_modified_global_path"   value="$(arg only_testing_modified_global_path)" />
    <param name="min_radius"   value="$(arg min_radius)" />
  </node>
</launch>
